pydantic_ai_slim/pydantic_ai/ag_ui.py

```diff
@@ -378,6 +398,15 @@ async def run_ag_ui(
                 run_id=run_input.run_id,
             ),
         )
+    finally:
+        # Call the callback if provided and we have a completed run
+        if on_complete is not None and completed_run is not None:
+            import inspect
+
+            if inspect.iscoroutinefunction(on_complete):
+                await on_complete(completed_run)
+            else:
+                on_complete(completed_run)
```

Comment:
Let's use `await _utils.run_in_executor(on_complete, completed_run)` like we do for all other user-registered functions.